crt_portal/static/js/dashboards-index-page.js

```diff
@@ -0,0 +1,182 @@
+      }
+    }
+  }
+
+  function handleDashboardsPaginationNextPrevious(section, forward = true) {
+    const dashboardDataBySection = JSON.parse(
+      document.getElementById('dashboards-by-section-data').textContent
+    );
+    const dashboardsList = dashboardDataBySection[section];
+
+    const sectionDashboardsCardsContainerElement = document.getElementById(
+      section + '-dashboards-cards-container'
+    );
+    const currentPageNumber = parseInt(sectionDashboardsCardsContainerElement.dataset.currentPage);
+    const pageSize = parseInt(sectionDashboardsCardsContainerElement.dataset.pageSize);
+
+    if (forward) {
+      paginateSectionDashboards(section, dashboardsList, pageSize, currentPageNumber + 1);
+    } else {
+      paginateSectionDashboards(section, dashboardsList, pageSize, currentPageNumber - 1);
+    }
+  }
+
+  function handleDashboardsPaginateToPage(section, page) {
+    const dashboardDataBySection = JSON.parse(
+      document.getElementById('dashboards-by-section-data').textContent
+    );
+    const dashboardsList = dashboardDataBySection[section];
+
+    const sectionDashboardsCardsContainerElement = document.getElementById(
+      section + '-dashboards-cards-container'
+    );
+    const pageSize = parseInt(sectionDashboardsCardsContainerElement.dataset.pageSize);
+    paginateSectionDashboards(section, dashboardsList, pageSize, page);
+  }
+
+  function setupSectionPaginationButtons(section) {
+    const paginationPrevious = document.getElementById(section + '-dashboards-previous');
+    paginationPrevious.addEventListener('click', () => {
+      handleDashboardsPaginationNextPrevious(section, false);
+    });
+
+    const paginationFirst = document.getElementById(section + '-dashboards-first-page');
+    paginationFirst.addEventListener('click', () => {
+      handleDashboardsPaginateToPage(section, 0);
+    });
+
+    const sectionDashboardsCardsContainerElement = document.getElementById(
+      section + '-dashboards-cards-container'
+    );
+    const paginationLast = document.getElementById(section + '-dashboards-last-page');
+    paginationLast.addEventListener('click', () => {
+      handleDashboardsPaginateToPage(
+        section,
+        parseInt(sectionDashboardsCardsContainerElement.dataset.lastPage)
+      );
+    });
+
+    const paginationNext = document.getElementById(section + '-dashboards-next');
+    paginationNext.addEventListener('click', () => {
+      handleDashboardsPaginationNextPrevious(section);
+    });
+  }
+
+  function populateInitialSectionDashboards() {
+    const dashboardDataBySection = JSON.parse(
+      document.getElementById('dashboards-by-section-data').textContent
+    );
+    const containers = document.getElementsByName('section-dashboards-cards-container');
+    for (const cardsContainerElement of containers) {
+      const sectionName = cardsContainerElement.id.split('-')[0];
+      if (sectionName in dashboardDataBySection) {
```

Comment:
careful - `in` [might not do what you want here](https://stackoverflow.com/a/13633027).

Prefer `hasOwnProperty` instead.

```suggestion
      if (dashboardDataBySection.hasOwnProperty(sectionName)) {
```